### PR TITLE
Initial commit of Troubleshooting MiQ Authentication blog post

### PR DESCRIPTION
**Pull request by jvlcek**
Add two related blog posts. One provides a high level overview
of MiQ Authentication technologies. The second provides some
troubleshooting tips.

https://www.pivotaltracker.com/n/projects/1610127/stories/154089398

Many issues reported against MiQ Authentication turn out to be basic missunderstandings
of what is supported and how to find the correct configuration information. Also many
issues are reported with insufficient or incorrect data. The intention of these blog posts
to attempt to mitigate these issues.

It is hoped that more troubleshooting information can be added over time.

### CODE REVIEW

**Comment by jvlcek** [comment]
@miq-bot assign @abellotti

**Comment by jvlcek** [comment]
@miq-bot label blog post

**Comment by jvlcek** [comment]
@gtanzillo,  @abellotti and @cybette Please review.

**Comment by abellotti** [comment]
Thanks @jvlcek for writing this up!! before we dive into the details, can I bother you for a global s/MiQ/ManageIQ/ ?

**Comment by jvlcek** [comment]
@abellotti  MiQ -> ManageIQ done :) Thank you.

**Comment by jvlcek** [comment]
Thank you @djberg96 for catching some spelling/typo errors!

**Comment by djberg96** [comment]
👍 Looks great!

File: site/_posts/2018-01-28-troubleshooting-auth.md [hunk 1]
```diff
@@ -0,0 +1,440 @@
+--
```
Comment (cybette) [inline site/_posts/2018-01-28-troubleshooting-auth.md]: Need 3 dashes for the front matter

File: site/_posts/2018-01-28-troubleshooting-auth.md [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+title: "Troubleshooting ManageIQ Authentication"
+author: jvlcek
+date: 2018-01-28
```
Comment (cybette) [inline site/_posts/2018-01-28-troubleshooting-auth.md]: When would you like to publish this, next Monday perhaps? Please date accordingly, thanks!

File: site/_posts/2018-01-29-auth-overview.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
+---
+title: "ManageIQ Authentication Overview"
+author: jvlcek
+date: 2018-01-29
```
Comment (cybette) [inline site/_posts/2018-01-29-auth-overview.md]: You might want to date the Overview post before the Troubleshooting one (or on the same day is fine as well)

File: site/_posts/2018-01-28-troubleshooting-auth.md (hunk 2 above)
Comment (jvlcek) [reply site/_posts/2018-01-28-troubleshooting-auth.md]: @cybette Thank you.

Can this be pushed today?

The render tool I use to develop will not allow for a date in the future. I will set it to a date closer to today.

**Comment by cybette** [comment]
@jvlcek the technical details are way over my head, but I'm sure the other experts will review those. Just a couple minor stuff I commented on, and ping me when all reviews are done and I can merge it. Thanks!!

File: site/_posts/2018-01-29-auth-overview.md (hunk 3 above)
Comment (jvlcek) [reply site/_posts/2018-01-29-auth-overview.md]: @cybette I want the overview to be listed on the `all blogs` page above the Troubleshooting post. The simplest way I found to do this is to have the date of the Overview post be ahead of the Troubleshooting post.

File: site/_posts/2018-01-28-troubleshooting-auth.md (hunk 2 above)
Comment (cybette) [reply site/_posts/2018-01-28-troubleshooting-auth.md]: Date them yesterday (Jan 31) and I'll merge them today.

File: site/_posts/2018-01-29-auth-overview.md (hunk 3 above)
Comment (cybette) [reply site/_posts/2018-01-29-auth-overview.md]: ah ok. you can add a timestamp e.g. `date: 2018-01-31 08:30:00`

better to use use different timestamps (which doesn't show on the blog post but will order them).

**Comment by jvlcek** [comment]
@cybette Thank you! I did a walk through with @abellotti and @gtanzillo for the technical content.

File: site/_posts/2018-02-01-auth-overview.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  1. Create groups on the ManageIQ appliance
+
+# External Authentication with SAML
+---------------------------------------------------------------------
+
+  Configuring ManageIQ for external authentication using SAML is documented [here](http://manageiq.org/docs/reference/latest/auth/saml)
+
+  The identity provider tool, [Keycloak](http://www.keycloak.org/), has been used to test the SAML functionality.
+
+  A summary of how to configure an ManageIQ appliance to work with SAML is as follows:
```
Comment (abellotti) [inline site/_posts/2018-02-01-auth-overview.md]: ... configure _a_ ManageIQ ...

File: site/_posts/2018-01-31-troubleshooting-auth.md [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+  ManageIQ configuration: **Mode: Database**, relies on all **User**, **Group** and **Role** information being stored
+  directly in the ManageIQ database to support authentication and authorization.
+
+  + See: [**Capturing Authentication Data From ManageIQ DB**](#capturing-authentication-data-from-manageiq-db) for tips
+    on how to capture the authentication information stored in the ManageIQ Database.
+
+### **Mode: LDAP and Mode: LDAPS**
+
+  ManageIQ configurations: **Mode: LDAP and Mode: LDAPS**, are implemented with the MiqLdap driver, which queries LDAP
+  IdP servers directly via the [**net-ldap Rubygem**](https://rubygems.org/gems/net-ldap/).
```
Comment (abellotti) [inline site/_posts/2018-01-31-troubleshooting-auth.md]: drop IdP here, just LDAP servers.

File: site/_posts/2018-01-31-troubleshooting-auth.md [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+  miq_bash> /var/www/miq/vmdb/bin/rails runner 'puts Settings.authentication'
+  </code></pre>
+
+  + This shell command can be used to query the database for the list of known ***groups***:
+
+  + The groups returned from the IdP for users expected to be authenticated on ManageIQ must match at least one
+    group reported by this command. If not one must be manually added using the ManageIQ UI.
+
+  <pre><code style="white-space: pre">
+  miq_bash> /var/www/miq/vmdb/bin/rails runner 'MiqGroup.all.map { |u| puts "  #{u[:description]}" }'
```
Comment (abellotti) [inline site/_posts/2018-01-31-troubleshooting-auth.md]: the following may be easier to type:

```
MigGroup.pluck(:description)
```

File: site/_posts/2018-01-31-troubleshooting-auth.md [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+  </code></pre>
+
+  + This shell command can be used to query the database for the list of known ***users***:
+
+  + ManageIQ will auto create authenticated and authorized users if **Get User Groups** from the IdP is configured on the ManageIQ Authentication page.
+
+  <pre><code style="white-space: pre">
+  miq_bash> /var/www/miq/vmdb/bin/rails runner 'User.all.map { |u| puts "  #{u[:userid]}" }'
```
Comment (abellotti) [inline site/_posts/2018-01-31-troubleshooting-auth.md]: Same as above:

```
User.pluck(:userid)
```

File: site/_posts/2018-01-31-troubleshooting-auth.md (hunk 5 above)
Comment (jvlcek) [reply site/_posts/2018-01-31-troubleshooting-auth.md]: 👍

File: site/_posts/2018-02-01-auth-overview.md (hunk 4 above)
Comment (jvlcek) [reply site/_posts/2018-02-01-auth-overview.md]: 👍

**Comment by abellotti** [comment]
LGTM @jvlcek. Thanks for last round of updates. 👍

**Pull request merged**